superset/migrations/versions/2024-08-13_15-27_e53fd48cc078_remove_sl_dataset_users.py

```diff
@@ -34,9 +34,14 @@
 
 def upgrade():
     connection = op.get_bind()
-    if connection.dialect.name != "sqlite":
-        drop_fks_for_table("sl_dataset_users")
-    op.drop_table("sl_dataset_users")
+
+    try:
+        if connection.dialect.name != "sqlite":
+            drop_fks_for_table("sl_dataset_users")
+        op.drop_table("sl_dataset_users")
+    except sa.exc.NoSuchTableError:
```

Comment:
@sadpandajoe Would you mind adding a method called `has_table` to https://github.com/apache/superset/blob/master/superset/migrations/shared/utils.py to be reused in these and future migrations?

```
def has_table(table_name: str) -> bool:
    """
    Check if a table exists in the database.

    :param table_name: The table name
    :returns: True iff the table exists
    """

    insp = inspect(op.get_context().bind)
    return insp.has_table(table_name)
```

Reply:
yes, and/or change `drop_fks_for_table` to receive a `fail_when_missing=False`

Reply:
Even if we updated `drop_fks_for_table` we'd have to write something to drop table too, else it'll fail won't it?

Reply:
Whatever makes sense, maybe `drop_table(table_name, including_fks=[], only_if_it_exists=True)` or maybe it's for another PR, but a common issue is a migration that fails at step 2 and when you rerun it it fails at step one since step one can only run once (say a create table). Personally think we could use more of the "drop-table-but-only-if-it-exists" and "create-table-but-drop-it-first-if-it-exists" type utils to avoid getting people tangled up. We've had many people open issue with the message of "step 1 one of the migration fails, here's the stacktrace", only to realize that the issue was with a subsequent step.